src/Val.cc

```diff
@@ -938,10 +938,32 @@ static std::variant<ValPtr, std::string> BuildVal(const rapidjson::Value& j, con
         }
 
         case TYPE_INTERVAL: {
-            if ( ! j.IsNumber() )
-                return mismatch_err();
+            if ( j.IsNumber() )
+                return make_intrusive<IntervalVal>(j.GetDouble());
+
+            if ( j.IsString() ) {
+                auto parts = util::split(j.GetString(), " ");
+                if ( parts.size() != 2 )
+                    return "wrong interval format, must be 'value {day|hr|min|sec|msec|usec}'";
```

Comment:
For the purpose of the storage framework, I think `to_json()` should render intervals as numbers/doubles instead. If this means adding a new parameter (e.g. intervals_as_double=T or some more generic/extensible options record/struct).

Contrast, e.g., with `time` which doesn't render into ISO format, but instead unix timestamp in seconds.

For interoperability with external system, the verbose string representation makes little sense - consumers would need to parse this explicitly. Using double/float is less annoying.

Reply:
The parsing also needs to be more more fancy :-(

```
$ zeek -e 'print double_to_interval(2*24*3600 + 2*3600 +  2*60 + 2*1.0 + 2*0.1 + 2*0.0001)'
2.0 days 2.0 hrs 2.0 mins 2.0 secs 200.0 msecs 200.000021 usecs
```

I think it makes sense to have this within Zeek still, but lets try to avoid putting the burden on others, too?

Reply:
> The parsing also needs to be more more fancy :-(

This is fixed/tested in the first two fixup commits

> For the purpose of the storage framework, I think to_json() should render intervals as numbers/doubles instead. If this means adding a new parameter (e.g. intervals_as_double=T or some more generic/extensible options record/struct).

The problem with this is that `to_json` just calls `Val::BuildJSON` underneath via `Val::ToJSON`. Adding a type-specific flag to all of those methods breaks the generic-ness of that whole call-chain. I could see making a new version of `to_json` that takes a record of options and passing a `RecordVal` through to the C++, but I worry about having to do a bunch of repeat record lookups on a potential logging hotpath. Unfortunately the only other easy way to do this - using a `global` flag - is a no-go as well.

I made the most recent changes to `BuildJSON` to use rapidjson clear back in 3.0, but the original decision to log the broken-out version of `interval` goes back to `json.bro` before then.